src/libs/antares/study/parts/hydro/container.h

```diff
@@ -34,6 +34,98 @@
 
 namespace Antares::Data
 {
+/*!
+ ** \brief Final Reservoir Level data for a single area
+ */
+class FinalReservoirLevelRuntimeData
+{
+public:
+    // FInal reservoir level runtime data
+    std::vector<bool> includeFinalReservoirLevel;
+    std::vector<double> endLevel;
+    std::vector<double> deltaLevel;
+
+    // Simulation Data
+    uint simEndDay;
+
+    // simulation year
+    uint yearIndex;
+
+    // data per area
+    double initialReservoirLevel;
+    double finalReservoirLevel;
+    double deltaReservoirLevel;
+    double reservoirCapacity;
+    double lowLevelLastDay;
+    double highLevelLastDay;
+    int initReservoirLvlMonth;
+
+    // methods:
+
+    void fillEmpty()
+    {
+        includeFinalReservoirLevel.push_back(false);
+        endLevel.push_back(0.);
+        deltaLevel.push_back(0.);
+    }
+
+    void assignEndLevelAndDelta()
+    {
+        includeFinalReservoirLevel.at(yearIndex) = true;
+        endLevel.at(yearIndex) = finalReservoirLevel;
+        deltaLevel.at(yearIndex) = deltaReservoirLevel;
+    }
+
+    double calculateTotalInflows(Antares::Memory::Stored<double>::Type& srcinflows) const
```

Comment:
This function already exists. Re-use it ?

Reply:
actually, as I can see it does not exist. 
Total inflows are calculated in method prepareInflowsScaling in management.cpp. 
It is a big method doing a lot of additional stuff, not good for re-using.